tests/yam/migration/migration_unattended.pm

```diff
@@ -0,0 +1,58 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Migration activation then reboot to perform migration.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use grub_utils 'grub_test';
+use migration 'disable_installation_repos';
+use power_action_utils 'power_action';
+use utils 'zypper_call';
+
+sub await_migration {
+    my $timeout = 600;
+    my @tags = qw(migration-exit grub2);
+    while (1) {
+        my $ret = check_screen \@tags, 30;
+	$timeout -= 30;
+        diag("left total await_migration timeout: $timeout");
+	if (match_has_tag('grub2')) {
+            last;
+        }
+	if (match_has_tag('migration-exit')) {
+            wait_still_screen 30;
+            next;
+        }
+    }
```

Comment:
wondering how we could simplify this, ideally we could do:
- assert needle migration-running (small timeout)
- assert needle migration-not-running (longer timeout)

Reply:
what I mean it is that I would prefer less complex logic if possible, without `while` or multiple matches. Instead of the the loop just create needle for sles 16 grub, you could probably simulate a failure not following the requirements described in confluence to rollback and create the needle of the grub for SP7 (although you could do it directly using another job tool).